metecho/api/views.py

```diff
@@ -198,6 +213,61 @@ class UserViewSet(mixins.RetrieveModelMixin, mixins.ListModelMixin, GenericViewS
     queryset = User.objects.all()
 
 
+class ProjectDependencyViewSet(ReadOnlyModelViewSet):
+    """Dependencies available during Project creation"""
+
+    permission_classes = (IsAuthenticated,)
+    serializer_class = ProjectDependencySerializer
+    queryset = ProjectDependency.objects.all()
+
+
+class GitHubOrganizationViewSet(ReadOnlyModelViewSet):
+    """GitHub Organizations available during Project creation"""
+
+    permission_classes = (IsAuthenticated,)
+    serializer_class = GitHubOrganizationSerializer
+    pagination_class = CustomPaginator
+    queryset = GitHubOrganization.objects.all()
+
+    @extend_schema(request=None, responses=ShortGitHubUserSerializer(many=True))
+    @action(detail=True, methods=["GET"])
+    def members(self, request, pk):
+        """Fetch the members of an Organization from GitHub"""
+        org: GitHubOrganization = self.get_object()
+        gh_api = gh.gh_given_user(request.user)
+        gh_org = gh_api.organization(org.login)
+        members = sorted(
+            (member.as_dict() for member in gh_org.members()),
+            key=lambda member: member["login"].lower(),
+        )
+        members = ShortGitHubUserSerializer(data=members, many=True)
+        members.is_valid(raise_exception=True)
+        return Response(members.data)
```

Comment:
Note that this list could be very long. If github3 doesn't paginate for us, should we do pagination on our end?

Reply:
Interestingly github3py actually makes our lives harder in this instance because it hides the GitHub API pagination in the call to `gh_org.members()` to return a seamless iterator. We can bypass that by calling the GitHub API directly with a page number read from GET params. Working on that now

Reply:
Added pagination in 4632f282d89abe230e2adbe8e338569f0e31e70b

Reply:
I want to be sure I understand the changes here. When we used `gh_org.members()` before, was that paginated or limited at all? Or was that fetching the entire list, no matter how long?

I think this is a good idea, to paginate this list. We may also want to add a new Trello card to consider paginating the `project.github_users` list, but that sounds more complex and maybe not a high priority if it hasn't presented as a concern yet.

My main question here is how the FE is supposed to know whether this list of `members` has been paginated or not? That is, how do we know whether there is a second page?

Reply:
`gh_org.members()` let's you [iterate over lists of objects as a continuous list](https://github3py.readthedocs.io/en/master/api-reference/structs.html), even though it actually fires multiple requests to the GH API behind the scenes. By calling the GH API endpoint directly we get access to the objects page by page.

> That is, how do we know whether there is a second page?

Huh, it looks like there's no way to get this information from the GitHub API. The [`/members` endpoint](https://docs.github.com/en/rest/orgs/members#list-organization-members) returns a plain list, and there's nothing related to membership count in the actual [`/organization/<id>` endpoint](https://docs.github.com/en/rest/orgs/orgs#get-an-organization). We could try using `org["plan"]["filled_seats"]` but that seem brittle.

The only way to be completely sure is requesting for another page until an empty list is returned (a "load more" pattern instead of actual pagination) 😕

Reply:
I guess we could also fast-track the story that creates an actual `GitHubUser` model and just filter and paginate a queryset of that instead. Alternatively we can also cache the full list returned by `gh_org.members()` and return that.

Reply:
I think my leaning would be to revert to the original `gh_org.members()` here, and then add a new story (or comment on the GitHubUser model story) to consider pagination options. What do you think?

Reply:
Sounds good 👍

Reply:
@jerivas Reminder to revert the members list pagination 🙏

Reply:
Sorry! Reverted in 8d8e5e9a462a15e3c532e3e0e9f4a329ce99a621